### ISSUE TITLE
indicator of unsaved configuration changes

### ISSUE THREAD

**Issue by ksu-0** [weechat/weechat#1694]
It would be nice if there was a way to inform the user of unsaved configuration changes, as well as a way to list them.

**Comment by weechatter** [comment]
you can easily write an script for it. for me, not that important to have this function in core.

**Comment by ksu-0** [comment]
I can see how to check that a configuration option is changed (with `hook_config`), but not how to check that it is saved. Would you be so kind to tell?

**Comment by weechatter** [comment]
yes, of course. you have to check with hook_command_run("/save").
WeeChat does not save config automatically. you have to either run "/save" manually or create a trigger that will run the "/save" command.

**Comment by ksu-0** [comment]
There is also `config_write`, but I guess false positives don't matter much here. Thank you, I will use this.

**Comment by flashcode** [comment]
This is not so easy to implement, even in WeeChat itself, because some sections of configuration files are dynamic and don't have real option (that you can /set), for example: filters, key bindings, and others in plugin config files.

So for example if you add a key binding (`/key bind ctrl-Z ...`), this can be considered as unsaved change because on next `/save` this key will be added to weechat.conf. But there is no way to know that such changes are pending.

Ideally all options written to config files have a suitable option in memory, and then I could add a flag in config files to know if they are "clean" or "dirty" (with pending changes not yet written to disk).

**Comment by ksu-0** [comment]
As far as I can tell, the following configuration is invisible to `hook_config`:

Autosaved on change:
* secured data

Sending a signal on change:
* filters – `filter_added`, `filter_removed`
* key bindings – `key_bind`, `key_unbind`

Impossible to handle:
* ignores (removals send a signal, additions do not)
* layouts
